internal/mockstore/mocktikv/pd.go

```diff
@@ -465,24 +471,194 @@ func (c *pdClient) Put(ctx context.Context, key []byte, value []byte, opts ...op
 	return nil, nil
 }
 
-func (m *pdClient) LoadResourceGroups(ctx context.Context) ([]*rmpb.ResourceGroup, int64, error) {
+func (c *pdClient) LoadResourceGroups(ctx context.Context) ([]*rmpb.ResourceGroup, int64, error) {
 	return nil, 0, nil
 }
 
-func (m *pdClient) GetServiceDiscovery() sd.ServiceDiscovery { return nil }
+func (c *pdClient) GetServiceDiscovery() sd.ServiceDiscovery { return nil }
 
-func (m *pdClient) WithCallerComponent(caller.Component) pd.Client { return m }
+func (c *pdClient) WithCallerComponent(caller.Component) pd.Client { return c }
 
-func (m *pdClient) GetGCInternalController(keyspaceID uint32) gc.InternalController {
-	return nil
+func enforceCircuitBreakerFor(name string, ctx context.Context) {
+	if circuitbreaker.FromContext(ctx) == nil {
+		panic(fmt.Errorf("CircuitBreaker must be configured for %s", name))
+	}
 }
 
-func (m *pdClient) GetGCStatesClient(keyspaceID uint32) gc.GCStatesClient {
-	return nil
+func (c *pdClient) GetGCInternalController(keyspaceID uint32) pdgc.InternalController {
+	return &gcInternalController{
+		inner:      c,
+		keyspaceID: keyspaceID,
+	}
 }
 
-func enforceCircuitBreakerFor(name string, ctx context.Context) {
-	if circuitbreaker.FromContext(ctx) == nil {
-		panic(fmt.Errorf("CircuitBreaker must be configured for %s", name))
+func (c *pdClient) GetGCStatesClient(keyspaceID uint32) pdgc.GCStatesClient {
+	return &gcStatesClient{
+		inner:      c,
+		keyspaceID: keyspaceID,
+	}
+}
+
+type gcInternalController struct {
+	inner      *pdClient
+	keyspaceID uint32
+}
+
+func (c gcInternalController) AdvanceTxnSafePoint(ctx context.Context, target uint64) (pdgc.AdvanceTxnSafePointResult, error) {
+	if c.keyspaceID != constants.NullKeyspaceID {
+		panic("unimplemented")
```

Comment:
???

Reply:
The `mocktikv` is an old mock implementation of TiKV that we've considered to remove it many times, but it's still here due to complexity and heaviness of the work, and still used by some tests. It's very unlikely that it will be used by any new tests where keyspaces are equipped. But the implementation will later be the same as the unistore one in [this PR](https://github.com/pingcap/tidb/pull/61529).